src/claim_signer/GetClaimSigner.ts

```diff
@@ -0,0 +1,151 @@
+/* tslint:disable:no-relative-imports */
+import * as crypto from 'crypto'
+import * as cuid from 'cuid'
+import JSONLD = require('jsonld')
+import JSONLD_SIGS = require('jsonld-signatures')
+import * as ParseDataUrl from 'parse-data-url'
+
+import { IllegalArgumentException } from '../Exceptions'
+import { Claim, ClaimContext, ClaimType, claimTypeDefaults, DefaultClaimContext } from '../Interfaces'
+import { getBase58ED25519PublicKeyFromPrivateKey } from '../util/KeyHelper'
+
+export interface ClaimSigner {
+  readonly createClaim: (type: ClaimType, claimAttributes: object, context: ClaimContext) => Claim
+}
+
+export const getClaimSigner = () => {
+  const jsonld = JSONLD()
+
+  const getDataDocumentLoader = (jsonld: any) => {
```

Comment:
If `getDataDocumentLoader` needs the `jsonld` instance, it could just grab it from the outer scope, which would allow us to simplify the signature and remove the wrapping function altogether.

```ts
export const getClaimSigner = () => {
  const jsonld = JSONLD()
  const nodeDocumentLoader = jsonld.documentLoaders.node({ usePromise: true })
  const dataDocumentLoader = async (url: string, callback: (error: any, data: any) => any) => {
      ...
  }
  jsonld.documentLoader = dataDocumentLoader
}
```

But `jsonld.documentLoaders.node` seems to be [a library constant](https://github.com/digitalbazaar/jsonld.js/blob/master/lib/jsonld.js#L944-L946)... if its meant to never change, we could just grab it from `JSONLD` instead of `jsonld`, which would allow us to have `const dataDocumentLoader` living in global space, not depending on any state.

Just for fun, we can even `const nodeDocumentLoader = require('jsonld/lib/documentLoaders/node.js')`, but that doesn't look very nice / future proof.

@kennylavender @krobi64 please let me know if this works / goes well with the implementation

Reply:
> If `getDataDocumentLoader` needs the `jsonld` instance...

What's wrong with making that dependency explicit in the function signature?